xikolo-ios/ViewControllers/Courses/CourseListViewController.swift

```diff
@@ -31,7 +33,21 @@ class CourseListViewController : AbstractCourseListViewController {
         updateView()
     }
 
-    override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
+    func detectSizeClass() {
+        switch traitCollection.horizontalSizeClass {
+        case UIUserInterfaceSizeClass.Compact, UIUserInterfaceSizeClass.Unspecified:
+            numberOfItemsPerRow = 1
+        case UIUserInterfaceSizeClass.Regular:
+            numberOfItemsPerRow = 2
+        }
+    }
+
+    override func traitCollectionDidChange(previousTraitCollection: UITraitCollection?) {
+        detectSizeClass()
```

Comment:
Why move the code to a separate function? It's only called once and `traitCollectionDidChange` only calls that function. Just leave the code here.